src/dynamic/HOmod_generator.cpp

```diff
@@ -42,33 +42,28 @@
 using TIME = float;
 
 // Ports for coupled models, we use the same in every level
-struct coupled_in_port1 : public cadmium::in_port<int>{};
-struct coupled_in_port2 : public cadmium::in_port<int>{};
-struct coupled_out_port : public cadmium::out_port<int>{};
-cadmium::dynamic::modeling::Ports coupled_in_ports = {typeid(coupled_in_port1), typeid(coupled_in_port2)};
-cadmium::dynamic::modeling::Ports coupled_out_ports = {typeid(coupled_out_port)};
-
-//A configured version of the devstone atomic, we use same configuration in every atomic.
-template<typename T>
-struct configured_atomic_devstone : devstone_atomic<T>{
-    configured_atomic_devstone(){
-        devstone_atomic<T>::period = 1;
-        devstone_atomic<T>::external_cycles = 100;
-        devstone_atomic<T>::internal_cycles = 100;
-    }
-};
+struct coupledHOmod_in_port1 : public cadmium::in_port<int>{};
+struct coupledHOmod_in_port2 : public cadmium::in_port<int>{};
+struct coupledHOmod_out_port : public cadmium::out_port<int>{};
 
 using ModelMatrix = std::vector<std::vector<std::shared_ptr<cadmium::dynamic::modeling::model>>>;
 
-std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> create_HOmod_model(uint width, uint depth) {
+std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> create_HOmod_model(
+        uint width, uint depth, int ext_cycles, int int_cycles, TIME time_advance) {
     // Creates the HOmod model with the passed parameters
     // Returns a shared_ptr to the TOP model
 
+    auto make_atomic_devstone = [&ext_cycles, &int_cycles, &time_advance](std::string model_id) -> std::shared_ptr<cadmium::dynamic::modeling::model> {
+        return cadmium::dynamic::translate::make_dynamic_atomic_model<devstone_atomic, TIME>(model_id, ext_cycles, int_cycles, time_advance);
+    };
     //Level 0 has always a single model
-    std::shared_ptr<cadmium::dynamic::modeling::model> devstone_atomic_L0_0 = cadmium::dynamic::translate::make_dynamic_atomic_model<configured_atomic_devstone, TIME>("devstone_atomic_L0_0");
+    std::shared_ptr<cadmium::dynamic::modeling::model> devstone_atomic_L0_0 = make_atomic_devstone("devstone_atomic_L0_0");
     std::unordered_map<int, ModelMatrix> atomics_by_level;
```

Comment:
Not sure using an unordered map would be the optimal collection in this case, it may add up unnecessary latency to the experiments. Lets look into this in a future iteration (doesn't have to be in the near future).